data/Store.js

```diff
@@ -697,15 +702,7 @@ export class Store {
         }
 
         // Note idSpec run against raw data here.
-        const id = this.idSpec(raw),
-            rec = this.getById(id);
-
-        if (rec) {
-            // We are creating a record for an update. Lookup our parent record and determine if
-            // this is the the summary record based on the current state (if not provided).
-            parent = withDefault(parent, rec.parent);
-            isSummary = withDefault(isSummary, rec.id === this.summaryRecord?.id);
```

Comment:
I think we also need to add this check up in the `updateData` method now like are doing for `parent` to ensure we are setting this flag if the summary record is provided in the update data?

Reply:
see later in the updateDate method where we intentionally identify and recreate the summary record, after doing a merge with the existing one, and at that point only set the flag